lib/std/os/linux.zig

```diff
@@ -3107,6 +3107,21 @@ pub const Sigaction = extern struct {
 
 pub const empty_sigset = [_]u32{0} ** @typeInfo(sigset_t).Array.len;
 
+pub const OffsetMask = struct {
+    offset: u8,
+    mask: u8,
+};
+
+// workaround, because @as(sigset_t, 1) << (SIG.XYZ - 1); does not work
```

Comment:
I really hate this workaround. Better suggestions?

Reply:
Something like `EnumSet`; though we should add something that assures it's dense.

Reply:
`x/os/socket_posix.zig` uses 
```zig
        pub fn init(domain: u32, socket_type: u32, protocol: u32, flags: std.enums.EnumFieldStruct(Socket.InitFlags, bool, false)) !Socket {
            var raw_flags: u32 = socket_type;
            const set = std.EnumSet(Socket.InitFlags).init(flags);
            if (set.contains(.close_on_exec)) raw_flags |= os.SOCK.CLOEXEC;
            if (set.contains(.nonblocking)) raw_flags |= os.SOCK.NONBLOCK;
            return Socket{ .fd = try os.socket(domain, raw_flags, protocol) };
        }
```
As I understand it, this only checks existence of an enum field. However, I would like ideally something that
1. checks comptime offset-safety and 2. does the shifting.

I could also add something to mem.zig, but I would like approval first to ensure there is no other usable solution.

Reply:
@daurnimator
```zig
        var sigabrtmask = std.bit_set.ArrayBitSet(u32, @typeInfo(sigset_t).Array.len);
        comptime sigabrtmask.initEmpty();
        comptime sigabrtmask.set(SIG.ABRT);
        sigprocmask(SIG.UNBLOCK, &sigabrtmask, null);
```
does the trick.